emscripten.py

```diff
@@ -306,14 +318,22 @@ def move_preasm(m):
   if shared.Settings.RELOCATABLE:
     global_funcs += ['g$' + extern for extern in metadata['externs']]
 
-  bg_funcs = basic_funcs + global_funcs
-  bg_vars = basic_vars + global_vars
+  basic_funcs = define_asmjs_import_names(basic_funcs, shared.Settings.MINIFY_ASMJS_IMPORT_NAMES)
+  global_funcs = define_asmjs_import_names(global_funcs, shared.Settings.MINIFY_ASMJS_IMPORT_NAMES)
+  basic_vars = define_asmjs_import_names(basic_vars, shared.Settings.MINIFY_ASMJS_IMPORT_NAMES)
+  global_vars = define_asmjs_import_names(global_vars, shared.Settings.MINIFY_ASMJS_IMPORT_NAMES)
+
+  def join_dicts(a, b):
+    return dict(a, **b)
```

Comment:
maybe @sbc100 has thoughts on the python style here?

Reply:
i'm not familiar with that syntax.   Its not clear how that is working.   I assume its the same as 

````
rtn = dict(a)
rtn.update(b)
return rtn
````

Is there any change the keys can collide here?